asdf/extension/_legacy.py

```diff
@@ -189,11 +199,11 @@ def get_cached_asdf_extension_list(extensions):
     # instances in identical order.
     extensions = tuple(ExtensionProxy.maybe_wrap(e) for e in extensions)
 
-    return _get_cached_asdf_extension_list(extensions)
+    return __get_cached_asdf_extension_list(extensions)
```

Comment:
I'm not a fan of a single prefixed `__` (unless name mangling is the desired outcome), even though here it may not name mangle as it can create confusion.